src/components/dataItem/BooleanStyle.js

```diff
@@ -10,52 +10,49 @@ const style = {
     marginTop: 20,
 }
 
-class BooleanStyle extends Component {
-    static propTypes = {
-        setBooleanStyle: PropTypes.func.isRequired,
-        valueType: PropTypes.string.isRequired,
-        values: PropTypes.shape({
-            true: PropTypes.string.isRequired,
-            false: PropTypes.string,
-        }),
-    }
-
-    componentDidMount() {
-        const { valueType, values, setBooleanStyle } = this.props
+const BooleanStyle = ({ valueType, values }) => {
+    const dispatch = useDispatch()
 
+    useEffect(() => {
         if (!values) {
-            setBooleanStyle('true', qualitativeColors[0])
+            dispatch(setBooleanStyle('true', qualitativeColors[0]))
 
             if (valueType === 'BOOLEAN') {
```

Comment:
Maybe a later refactor can switch these from strings to constants (in `constants/valueTypes.js`)